// This function can be used to get an identifying xpub at the keypath m/4541509'/1112098098'"
// The keypath argument has to be m/4541509'/1112098098'

syntax = "proto3";
package shiftcrypto.bitbox02;

import "google/protobuf/empty.proto";

message ElectrumEncryptionKeyRequest {
  repeated uint32 keypath = 1;
}

message ElectrumEncryptionKeyResponse {
  string key = 1;
}

message BIP85Request {
  message AppLn {
    uint32 account_number = 1;
  }

  oneof app {
    google.protobuf.Empty bip39 = 1;
    AppLn ln = 2;
  }
}

message BIP85Response {
  oneof app {
    google.protobuf.Empty bip39 = 1;
    bytes ln = 2;
  }
}

message UnlockRequest {
  // If true, the device will be allowed to ask the user to enter the passphrase on the host.  If
  // the user accepts, the host will receive a `UnlockRequestHostInfoResponse` with type
  // `PASSPHRASE`.
  bool supports_host_passphrase = 1;
}

message UnlockRequestHostInfoResponse {
  enum InfoType {
    UNKNOWN = 0;
    // Respond with `UnlockHostInfoRequest` containing the passphrase.
    PASSPHRASE = 1;
  }
  InfoType type = 1;
}

message UnlockHostInfoRequest {
  // Omit if type==PASSPHRASE and entering the passhrase is cancelled on the host.
  optional string passphrase = 1;
}
